File: dlr_ros_utils/scripts/time_example.py
#!/usr/bin/env python3

import rospy

if __name__ == '__main__':

    rospy.init_node("time_example", anonymous=False)

    start_time = rospy.Time.now()

    loop_rate = rospy.Rate(0.5)

    while not rospy.is_shutdown():
        rospy.loginfo("Actual time: " + str(rospy.Time.now()))
        rospy.loginfo("Elapsed time from start_time in seconds: " + 
                      str((rospy.Time.now() - start_time).to_sec()) + "\n")
        
        loop_rate.sleep()
  

File: dlr_ros_utils/src/time_example.cpp
#include <ros/ros.h>

int main (int argc, char** argv)
{
    ros::init(argc, argv, "time_example");

    ros::NodeHandle nh;

    ros::Time start_time = ros::Time::now();

    ros::Rate loop_rate(0.5);

    while(ros::ok())
    {
        ROS_INFO_STREAM("Actual time: " << ros::Time::now());
        ROS_INFO_STREAM("Elapsed time from start_time in seconds: " << (ros::Time::now() - start_time).toSec() << "\n");
        
        loop_rate.sleep();
    }

    return 0;
}

